nomad/testing.go

```diff
@@ -39,6 +39,14 @@ func TestACLServer(t *testing.T, cb func(*Config)) (*Server, *structs.ACLToken,
 }
 
 func TestServer(t *testing.T, cb func(*Config)) (*Server, func()) {
+	s, c, err := TestServerWithErr(t, cb)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	return s, c
+}
+
+func TestServerWithErr(t *testing.T, cb func(*Config)) (*Server, func(), error) {
```

Comment:
I'm not sure if this is the best approach. I need to test the server doesn't start, but it was always failing the test due to the `t.Fatalf`. 

I found this [other approach](https://blog.antoine-augusti.fr/2015/12/testing-an-os-exit-scenario-in-golang/), but it doesn't sound right for this case.

Reply:
Seems reasonable